pkg/jmespath/jmespath.go

```diff
@@ -0,0 +1,203 @@
+package jmespath
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"maps"
+	"os"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/buildbarn/bb-storage/pkg/program"
+	pb "github.com/buildbarn/bb-storage/pkg/proto/jmespath"
+	"github.com/buildbarn/bb-storage/pkg/util"
+	"github.com/jmespath/go-jmespath"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// Expression represents a parsed JMESPath expression.
+type Expression struct {
+	expression *jmespath.JMESPath
+	files      *metadataFileProvider
+}
+
+// NewExpressionFromConfiguration creates a new JMESPath expression.Expression
+// from the provided configuration. This will also evaluate all test vectors
+// and return an error if any of them fail.
+//
+// The group parameter can be optionally supplied: if it is not nil, then
+// any background refresher threads will be run using the group. Otherwise
+// background threads will be executed as a normal goroutine.
```

Comment:
I'd appreciate it if we used the occasion to plumb through the use of `program.Group` instead.

Reply:
I've done this, and have tried to be consistent with existing code in terms of the `program.Group` argument. It's touched quite a lot of places, in particular inside the blob access creator. This could have been avoided by putting the `program.Group` inside the `grpc.ClientFactory`, but this would have been a bit inconsistent with other existing code.